li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerConfig.java

```diff
@@ -100,7 +107,9 @@
         .define(MAX_MESSAGE_SEGMENT_BYTES_CONFIG, Type.INT, "800000", Importance.MEDIUM, MAX_MESSAGE_SEGMENT_BYTES_DOC)
         .define(AUDITOR_CLASS_CONFIG, Type.CLASS, NoOpAuditor.class.getName(), Importance.MEDIUM, AUDITOR_CLASS_DOC)
         .define(KEY_SERIALIZER_CLASS_CONFIG, Type.CLASS, ByteArraySerializer.class.getName(), Importance.MEDIUM, KEY_SERIALIZER_CLASS_DOC)
-        .define(VALUE_SERIALIZER_CLASS_CONFIG, Type.CLASS, ByteArraySerializer.class.getName(), Importance.MEDIUM, VALUE_SERIALIZER_CLASS_DOC)
+        .define(VALUE_SERIALIZER_CLASS_CONFIG, Type.CLASS, null, Importance.MEDIUM, VALUE_SERIALIZER_CLASS_DOC) //null by default!
```

Comment:
Why did you change the default for value serializer?

Reply:
the default partitioner hashes the key, which we know works. in that case i can avoid all the reflection calls at runtime - this way users only pay the cost for this if they need it

Reply:
unintended. thanks for catching this